## Task 1: Permutations and Combinations

### Problem:

We have 12 cups of tea, 6 with milk added first and 6 with tea added first. A person, claiming to have the special ability to distinguish which cups had the milk added first, is guessing the cups.

1. **Part 1**: We will calculate the probability that they select all 6 cups with the milk added first correctly.
2. **Part 2**: We will calculate the probability that they make at most one error (i.e., they correctly select at least 5 out of the 6 cups with the milk added first).

The person is assumed to be guessing randomly.


In [2]:
# Part 1

from math import comb

# Total number of combinations for selecting 6 cups from 12
total_combinations = comb(12, 6)

# There is only 1 correct combination where the person selects exactly the 6 correct cups
correct_combination = 1

# Probability of guessing all 6 correctly, in percentage
probability_correct = (correct_combination / total_combinations) * 100

print(f"Total combinations of 6 cups from 12: {total_combinations}")
print(f"Probability of guessing all 6 correctly: {probability_correct:.4f}%")


Total combinations of 6 cups from 12: 924
Probability of guessing all 6 correctly: 0.1082%


In [3]:
# Part 2

# Number of ways to select exactly 5 correct cups and 1 incorrect cup
correct_5_wrong_1 = comb(6,5) * comb(6,1)

# Total number of ways to select exactly 6 or at least 5 correct cups
correct_6_or_5 = correct_combination + correct_5_wrong_1

# Ptobability of making at most 1 error 
probability_one_error =(correct_6_or_5/total_combinations) * 100

print(f"Number of ways to select exactly 5 correct cups and 1 incorrect: {correct_5_wrong_1}")
print(f"Probability of guessing at least 5 correctly (making at most 1 error): {probability_one_error:.4f}%")

Number of ways to select exactly 5 correct cups and 1 incorrect: 36
Probability of guessing at least 5 correctly (making at most 1 error): 4.0043%


In [4]:
# Number of ways to select exactly 4 correct cups and 2 incorrect cups
correct_4_wrong_2 = comb(6, 4) * comb(6, 2)

# Probability of making exactly 2 errors
probability_two_errors = (correct_4_wrong_2 / total_combinations) * 100

print(f"Number of ways to select exactly 4 correct cups and 2 incorrect: {correct_4_wrong_2}")
print(f"Probability of guessing at least 4 correctly (making exactly 2 errors): {probability_two_errors:.4f}%")


Number of ways to select exactly 4 correct cups and 2 incorrect: 225
Probability of guessing at least 4 correctly (making exactly 2 errors): 24.3506%


### Would We Accept Two Errors?

If we accept two errors, the person needs to select at least 4 correct cups (meaning they can guess up to 2 cups incorrectly). The probability of guessing at least 4 correct cups by chance increases to **24.35%**. This is significantly higher than the probability of guessing all 6 cups correctly by chance (**0.11%**).

#### Statistical Significance:

In statistics, we often determine whether an outcome is due to random chance by comparing it to a **significance level** (often denoted as α, commonly set to 5% or 0.05). If a result has a probability greater than this threshold, we are less likely to consider it statistically significant because it could easily happen by random chance.

- Allowing two errors leads to a **24.35%** probability, which is much higher than the common **5% significance level**.
- This means that with such a high probability, the outcome is **not statistically significant** and could very easily occur by random guessing, which undermines the credibility of the person's supposed ability.

In contrast, if we require the person to guess all 6 cups correctly, the probability is **0.11%**—well below the 5% significance threshold. This low probability indicates that such a result would be **statistically significant**, meaning it’s unlikely to have occurred by chance alone.

#### Conclusion:

Allowing two errors makes the test less rigorous because the probability of passing the test increases significantly (24.35%), approaching random chance. This result is not statistically significant, so accepting two errors would make it much more likely that a person could pass the test by guessing alone, rather than demonstrating any real skill or ability.
